odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/Odnoklassniki.java

```diff
@@ -409,10 +437,41 @@ private void signParameters(final Map<String, String> params) {
         params.put(Shared.PARAM_SIGN, sig);
     }
 
-    public final void setOkListener(OkListener listener) {
+    public final void setOkListener(final OkListener listener) {
```

Comment:
@lintax Leaving activity may cause it to be destroyed by the Android OS. I personally observe that  on some Samsungs cheap phones. 

It may cause at least two problems:
1. If you remove listener from in `onPause` or `onStop`, every time you will come from OkAuthActivity activity back to MainActivity you will not have desired data 
2. If you will not remove listener and system will destroy activity:
 * Strong reference  to MainActivity  will still exist in SDK preventing JVM to free memory
 * If callback will be called it may crash an app because activity no longer in correct state


Main idea of the fork was to create deferred data store which will hold all necessary results  if UI which need them is not available